src/wl-copy.c

```diff
@@ -396,17 +396,11 @@ int main(int argc, char * const argv[]) {
         {"seat", required_argument, 0, 's'},
         {0, 0, 0, 0}
     };
-    while (1) {
-        int option_index;
-        const char *opts = "vhpnofct:s:";
-        int c = getopt_long(argc, argv, opts, long_options, &option_index);
-        if (c == -1) {
-            break;
-        }
-        if (c == 0) {
-            c = long_options[option_index].val;
-        }
-        switch (c) {
+    const char *opts = "vhpnofct:s:";
+    int opt;
+
+    while ((opt = getopt_long(argc, argv, opts, long_options, NULL)) != -1) {
```

Comment:
In general I prefer not to use C's *assign and compare as a single expression* idiom. Is there a reason to do this besides it being a bit shorter?

Reply:
This is a nice way of doing it :)

It seems to be that this is generally the preferred way.